src/safe_allocator.h

```diff
@@ -62,7 +66,8 @@ class SafeAllocator : public Allocator {
     ~SafeAllocator() override = default;
 
 private:
-    Allocator* raw_allocator_ = nullptr;
+    Allocator* const raw_allocator_ = nullptr;
```

Comment:
`raw_allocator_` can removed ?

Reply:
if the data is allocated by user defined allocator ,we suggest the data must not be changed by other allocators. the pointer is used for user defined allocator, not for default.

Reply:
OK, the `self_own_allocator_` is used for shared_ptr reference holder.
But actual there is no such strong owned semantics here.

Reply:
I changed `self_own_allocator` to `owned_allocator`. While there is indeed no "self-owned" semantics here, the purpose is to ensure that I also hold the allocator to prevent it from being destructed prematurely.